### PR TITLE
ENH: Add option to compute the Cramer-Rao Lower Bound.

### PR DESCRIPTION
**Pull request by arobert01**
Add option to obtain the Cramer-Rao Lower Bound (CRLB) when computing the photon counts using rtkSpectralForwardModelImageFilter. The CRLB corresponds to the lower bound of the variance of any unbiased estimator. Therefore, it could be used to estimate the variance that would be obtained after material decomposition.

### CODE REVIEW

File: include/rtkSpectralForwardModelImageFilter.hxx [hunk 1]
```diff
@@ -470,8 +471,20 @@ SpectralForwardModelImageFilter<DecomposedProjectionsType,
...
+        photon_counts.SetData(forward.data_block());
+        cost->SetMeasuredData(photon_counts);
+        cost->ComputeFischerMatrix(in);
+        output1It.Set(cost->GetCramerRaoLowerBound());
+      }
+      else
+      {
+        output1It.Set(
+          itk::VariableLengthVector<double>(cost->GetVariances(in).data_block(), this->m_NumberOfSpectralBins));
+      }
```
Comment (SimonRit) [inline include/rtkSpectralForwardModelImageFilter.hxx]: I'm not sure I understand this, it seems to disable the previous mode which was already for `m_IsSpectralCT` `true`. Was the behavior different before?

Comment (arobert01) [reply include/rtkSpectralForwardModelImageFilter.hxx]: The function `GetVariances` was not implemented for spectral CT. It was only returning meaningless results (see [here](https://github.com/RTKConsortium/RTK/blob/master/include/rtkProjectionsDecompositionNegativeLogLikelihood.h#L236-L243)). Furthermore, it was no possible to set the boolean `m_ComputeVariances` to true with the `rtkspectralforwardmodel` application.

Comment (SimonRit) [reply include/rtkSpectralForwardModelImageFilter.hxx]: Ok... Do you understand what it does for dual energy?
https://github.com/RTKConsortium/RTK/blob/master/include/rtkDualEnergyNegativeLogLikelihood.h#L126
I have a hard time understanding this code and the [commit message](https://github.com/RTKConsortium/RTK/commit/0b3cf921f8422a2f96604786e56143e376bebadb) does not help

Comment (arobert01) [reply include/rtkSpectralForwardModelImageFilter.hxx]: If I understood well, it is mentioned [here](https://github.com/RTKConsortium/RTK/blob/master/include/rtkDualEnergyNegativeLogLikelihood.h#L112) that the variance of the integrated energy is assumed to be equal to the mean. Then, the function [GetVariances](https://github.com/RTKConsortium/RTK/blob/master/include/rtkDualEnergyNegativeLogLikelihood.h#L125-L139) is computing this:
``` math
\int_0^\infty E S_{eff}(E)exp(-\sum_{\alpha=1}^{M=2} f_\alpha A_\alpha)dE
```
which correspond to the mean signal measured by an energy integrating detector. 
(This is my understanding but I'm not sure I'm right.)

**Comment by arobert01** [comment]
The last force-pushed improve the way the computation of the Cramer-Rao lower bound was implemented and add the computation of the variance also for spectral CT.

**Pull request merged**